### PR TITLE
refactor(relay): use single data structure to contain pubsub items

### PR DESCRIPTION
**Pull request by richard-ramos**
# Description
This arguably simplifies code by having a single source of truth for all data related to pubsub topics, as well as handling an error reported by @fbarbu15 in which a panic is triggered right after this logline:
```
2023-11-15T15:23:29.481Z	[34mINFO[0m	gowaku.node2.relay	relay/waku_relay.go:368	subscribing to	{"node": "16Uiu2HAmGNtM2rQ8abySFNhqPDFY4cmfAEpfo9Z9fD3NekoFR2ip", "pubsubTopic": "test", "contentTopics": []}
fatal error: concurrent map iteration and map write
```

### CODE REVIEW

File: waku/v2/protocol/relay/waku_relay.go [hunk 1]
```diff
@@ -383,13 +397,13 @@ func (w *WakuRelay) subscribe(ctx context.Context, contentFilter waku_proto.Cont
 			WithConsumerOption(params.dontConsume))
 
 		// Create Content subscription
-		w.topicsMutex.RLock()
-		if _, ok := w.contentSubs[pubSubTopic]; !ok {
-			w.contentSubs[pubSubTopic] = map[int]*Subscription{}
+		w.topicsMutex.Lock()
+		defer w.topicsMutex.Unlock()
```
[the remark this answers is not in the document]
Comment (richard-ramos) [reply waku/v2/protocol/relay/waku_relay.go:401]: good catch!

**Review by chaitanyaprem — APPROVED** [review]
LGTM

Seems like tests-with-race is failing which needs to be fixed before merging. Not sure if it is just a flaky test.

**Comment by richard-ramos** [comment]
```
[2023-11-21T06:54:19.822Z] --- FAIL: TestFilterSuite (96.38s)
[2023-11-21T06:54:19.822Z]     --- FAIL: TestFilterSuite/TestLargePayloadsUTF8 (20.40s)
[2023-11-21T06:54:19.822Z]         filter_test.go:197: 
[2023-11-21T06:54:19.822Z]             	Error Trace:	github.com/waku-org/go-waku/waku/v2/protocol/filter/filter_test.go:197
[2023-11-21T06:54:19.822Z]             	            				runtime/asm_amd64.s:1598
[2023-11-21T06:54:19.822Z]             	Error:      	test exceeded allocated time
[2023-11-21T06:54:19.822Z]             	Test:       	TestFilterSuite/TestLargePayloadsUTF8
[2023-11-21T06:54:19.822Z]         filter_test.go:208: 
[2023-11-21T06:54:19.822Z]             	Error Trace:	github.com/waku-org/go-waku/waku/v2/protocol/filter/filter_test.go:208
[2023-11-21T06:54:19.822Z]             	            				github.com/waku-org/go-waku/waku/v2/protocol/filter/filter_push_test.go:136
[2023-11-21T06:54:19.822Z]             	Error:      	Should be true
[2023-11-21T06:54:19.822Z]             	Test:       	TestFilterSuite/TestLargePayloadsUTF8
[2023-11-21T06:54:19.822Z] FAIL
```

@romanzac

**Comment by romanzac** [comment]
> ```
> [2023-11-21T06:54:19.822Z] --- FAIL: TestFilterSuite (96.38s)
> [2023-11-21T06:54:19.822Z]     --- FAIL: TestFilterSuite/TestLargePayloadsUTF8 (20.40s)
> [2023-11-21T06:54:19.822Z]         filter_test.go:197: 
> [2023-11-21T06:54:19.822Z]             	Error Trace:	github.com/waku-org/go-waku/waku/v2/protocol/filter/filter_test.go:197
> [2023-11-21T06:54:19.822Z]             	            				runtime/asm_amd64.s:1598
> [2023-11-21T06:54:19.822Z]             	Error:      	test exceeded allocated time
> [2023-11-21T06:54:19.822Z]             	Test:       	TestFilterSuite/TestLargePayloadsUTF8
> [2023-11-21T06:54:19.822Z]         filter_test.go:208: 
> [2023-11-21T06:54:19.822Z]             	Error Trace:	github.com/waku-org/go-waku/waku/v2/protocol/filter/filter_test.go:208
> [2023-11-21T06:54:19.822Z]             	            				github.com/waku-org/go-waku/waku/v2/protocol/filter/filter_push_test.go:136
> [2023-11-21T06:54:19.822Z]             	Error:      	Should be true
> [2023-11-21T06:54:19.822Z]             	Test:       	TestFilterSuite/TestLargePayloadsUTF8
> [2023-11-21T06:54:19.822Z] FAIL
> ```
> 
> @romanzac

Where is the log coming from ? Test timeout was already extended. I cannot reproduce...

func (s *FilterTestSuite) TestLargePayloadsUTF8() {

	s.ctx, s.ctxCancel = context.WithTimeout(context.Background(), 40*time.Second)
...	
}

**Pull request merged**